### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Flashlock**
Firebase accounts can be created. We can log in and log out. We can navigate through the app

### CODE REVIEW

**Review by NamiSaya — APPROVED** [review]

**Review by CyanPhoenix51 — APPROVED** [review]

**Review by CrossoutX — APPROVED** [review]

**Comment by Flashlock** [comment]
Unstructure the files in main, wait until all primary branches have been merged before restructure

**Review by RolyatVo — APPROVED** [review]

**Pull request merged**
